src/gtest/test_mempoollimit.cpp

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2019 The Zcash developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or https://www.opensource.org/licenses/mit-license.php 
+
+#include <gtest/gtest.h>
+#include <iostream>
+
+#include "arith_uint256.h"
+#include "mempoollimit.h"
+#include "utiltime.h"
+#include "utiltest.h"
+#include "transaction_builder.h"
+
+
+const uint256 TX_ID1 = ArithToUint256(1);
+const uint256 TX_ID2 = ArithToUint256(2);
+const uint256 TX_ID3 = ArithToUint256(3);
+
+TEST(MempoolLimitTests, RecentlyEvictedList_AddWrapsAfterMaxSize)
+{
+    RecentlyEvictedList recentlyEvicted(2, 100);
+    SetMockTime(1);
+    recentlyEvicted.add(TX_ID1);
+    recentlyEvicted.add(TX_ID2);
+    recentlyEvicted.add(TX_ID3);
+    // tx 1 should be overwritten by tx 3 due maxSize 2
+    EXPECT_FALSE(recentlyEvicted.contains(TX_ID1));
+    EXPECT_TRUE(recentlyEvicted.contains(TX_ID2));
+    EXPECT_TRUE(recentlyEvicted.contains(TX_ID3));
```

Comment:
Perhaps test that `TX_ID2` is the next to be evicted. (I can imagine a buggy queue that would pass the test as it stands.)

Reply:
This has now been added this as a separate test.